File: js/src/samples/builder/from_file.ts
import { RecordBuilder } from "@bloock/sdk"
import { Config } from "../../utils/config"
import { Sample } from "../../utils/sample"
import { readFileSync, writeFileSync } from "fs"
import { Logger } from "../../utils/logger"

Sample.run("builder_from_file", async (_: Config) => {
    let file = readFileSync("../resources/dummy.pdf");
    let record = await RecordBuilder
        .fromFile(file)
        .build();

    Logger.success(`Record was created successfully`)

    let hash = await record.getHash()
    if (hash != "43fa336d95e1634fee2d3031adc44ed9464472b94511af1facb87a1fee0b7e0e") {
        throw new Error("Unexpected hash received")
    }

    Logger.success(`Hash: ${hash}`);

    // we can get the file back if needed
    writeFileSync("../resources/output.pdf", record.retrieve());
    Logger.success(`File written`);
})

File: python/samples/builder/from_file.py
from utils.logger import Logger
from utils.sample import Sample
from utils.config import Config

from bloock.client.builder import RecordBuilder

def from_file(_: Config):
    data = bytes()
    # we can read a file as an array of bytes
    with open("sample.pdf", "rb") as file:
        data = bytes(file.read())

    # and build a record from it
    record = RecordBuilder.from_file(data).build()
    
    Logger.success('Record was created successfully')

    hash = record.get_hash()
    if hash != "43fa336d95e1634fee2d3031adc44ed9464472b94511af1facb87a1fee0b7e0e":
        raise Exception("Unexpected hash received")

    Logger.success(f'Hash: {hash}')

    # we can get the file back if needed
    with open("output.pdf", "wb") as file:
        file.write(record.retrieve())

Sample("builder_from_file", from_file)
